codesy/base/models.py

```diff
@@ -4,42 +4,87 @@
 from django.conf import settings
 from django.contrib.auth.models import AbstractUser
 from django.db import models
-from django.db.models.signals import pre_save
 from django.dispatch import receiver
-
 from allauth.account.signals import user_signed_up
+# from django.contrib.postgres.fields import JSONField
 
+from .stripe_events import StripeAccount
 
 EMAIL_URL = 'https://api.github.com/user/emails'
 stripe.api_key = settings.STRIPE_SECRET_KEY
 
 
 class User(AbstractUser):
     # TODO: remove this as we don't need or want to store it
-    stripe_cc_token = models.CharField(max_length=100, blank=True)
-    stripe_account_token = models.CharField(max_length=100, blank=True)
+    stripe_card = models.CharField(max_length=100, blank=True)
+    stripe_customer = models.CharField(max_length=100, blank=True)
+    stripe_bank_account = models.CharField(max_length=100, blank=True)
+    tos_acceptance_date = models.DateTimeField(null=True, blank=True)
+    tos_acceptance_ip = models.CharField(max_length=20, blank=True)
     USERNAME_FIELD = 'username'
 
     def get_gravatar_url(self):
         github_account = self.socialaccount_set.get(provider='github')
         return ("https://avatars3.githubusercontent.com/u/%s?v=3&s=96" %
                 github_account.uid)
 
+    def accepted_terms(self):
+        return self.tos_acceptance_date and self.tos_acceptance_ip
+
+    def account(self):
+        try:
+            codesy_account = StripeAccount.objects.get(user=self)
+            return codesy_account
+        except:
+            return False
```

Comment:
returning `False` again - could the calling code just get the `User` object back and check that it doesn't have a `codesy_account`